### PR TITLE
Availability Entry Mark II

### PR DESCRIPTION
**Pull request by sjmcculloch**
The availability screens have been rebuilt according to the screen mockups provided by Mel & Simon.

The most significant change is that time entry is now governed by fixed timeSegments available system wide (eventually these will be configurable at a group and perhaps schedule level). 

You can select many days and tap any of the timeSegments to enter availability. Tapping multiple times will allow you to specify the unavailable & urgent statuses. You can also bulk-edit multiple days at once.

Once a timeSegment has been selected, it will automatically grey out conflicting timeSegments. You can toggle schedule information by tapping the icon.

image

image

image

image

The existing calendar components from the old screens have been removed

### CODE REVIEW

File: client/src/components/DateTime/DateRange.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    const day = startOfWeek.add(i === 0 ? 0 : 1, 'days').unix();
+    const dayUnix = 24 * 60 * 60;
+    const endOfDay = parseInt(day, 0) + dayUnix;
+
+    let availType = NOT_SPECIFIED;
+    if (day < startTime || day > endTime) {
+      availType = OUT_OF_RANGE;
+    } else {
+      const results = timeSegments.filter(
+        timeSegment => day <= timeSegment.startTime && endOfDay >= timeSegment.endTime,
```
Comment (sdunster) [inline client/src/components/DateTime/DateRange.js]: woah that's tough to read - confused the shit out of me. Maybe see if the linter will let you do this:
```
timeSegment => (day <= timeSegment.startTime && endOfDay >= timeSegment.endTime),
```

File: client/src/screens/availability/Detail.js [hunk 2]
```diff
@@ -8,61 +8,198 @@ import Icon from 'react-native-vector-icons/FontAwesome';
...
+
+  onShowInfo = () => {
+    this.setState({
+      showInfo: !this.state.showInfo,
+    });
+  };
+
+  getNextSegment = (segment) => {
+    switch (segment) {
+      case '':
```
Comment (sdunster) [inline client/src/screens/availability/Detail.js]: lets make a constant for `''` as well.

File: client/src/selectors/schedules.js [hunk 3]
```diff
@@ -1,5 +1,32 @@
 import moment from 'moment';
 
+import Colors from '../themes/Colors';
+
+export const isSelectorDisabled = (selectionSegments, status, startTime, endTime) => {
```
Comment (sdunster) [inline client/src/selectors/schedules.js]: it'd be so great to get some tests on this

File: client/src/selectors/schedules.js [hunk 4]
```diff
@@ -23,6 +50,19 @@ export const selectSchedules = (schedules, { startTime, endTime }) => {
   return filteredItems.sort((a, b) => (a.startTime > b.startTime ? 1 : -1));
 };
 
+export const selectColor = (status) => {
+  switch (status) {
```
Comment (sdunster) [inline client/src/selectors/schedules.js]: faster, easier test coverage:
```
const colors = {
  'Available': Colors.bgBtnAvailable,
  'Unavailable': Colors.bgBtnUnavailable,
  'Urgent': Colors.bgBtnUrgent,
};
const color = colors[status];
if (typeof colors === 'undefined') {
  return Color.bgWhite;
}
return color;
```
now you don't need to get coverage on every branch in the switch.

File: client/src/components/DateTime/DateRange.js (hunk 1 above)
Comment (sjmcculloch) [reply client/src/components/DateTime/DateRange.js]: The auto-formatter adjusts it to that unfortunately.

File: client/src/screens/availability/Detail.js (hunk 2 above)
Comment (sjmcculloch) [reply client/src/screens/availability/Detail.js]: Updated

File: client/src/selectors/schedules.js (hunk 3 above)
Comment (sjmcculloch) [reply client/src/selectors/schedules.js]: Agreed - that's why I put the logic in this area

File: client/src/selectors/schedules.js (hunk 4 above)
Comment (sjmcculloch) [reply client/src/selectors/schedules.js]: Updated, 

export const selectColor = (status) => {
  const colors = {
    Available: Colors.bgBtnAvailable,
    Unavailable: Colors.bgBtnUnavailable,
    Urgent: Colors.bgBtnUrgent,
  };
  const color = colors[status];
  if (typeof colors === 'undefined') {
    return Colors.bgWhite;
  }
  return color;
};

**Pull request merged**